hive/lib/src/box/box_base_impl.dart

```diff
@@ -163,6 +163,21 @@ abstract class BoxBaseImpl<E> implements BoxBase<E> {
     return delete(keystore.getAt(index)!.key, notify: notify);
   }
 
+  @override
+  Future<void> deleteAllAt(
+    Iterable<int> indexes, {
+    bool notify = true,
+  }) async {
+    for (var index in indexes.toList()..sort((a, b) => b.compareTo(a))) {
+      try {
+        deleteAt(index, notify: notify);
+      } catch (e) {
+        // Ignore errors
+      }
+    }
+    return;
```

Comment:
We already have `deleteAll` method. I would suggest resolving keys using the indices and then passing them to that method instead of calling deleteAt one by one.

Reply:
I've removed the sorting and any additional bulk.

Instead of calling `deleteAll`, I've used writeFrames, which is probably significantly more efficient than re-verifying that the key exists.

Reply:
Not sure if you would consider this "resolved." Please let me know if you think that using `deleteAll()` is a better method than using `writeFrames()`, and I'll implement the fix right away.